### PR TITLE
Deprecate Cluster resource migration guide

### PR DESCRIPTION
**Pull request by JakeSCahill**
## Description

Part of https://github.com/redpanda-data/redpanda-operator/pull/237

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/upgrade/pages/deprecated/index.adoc
Comment (chrisseto) [inline modules/upgrade/pages/deprecated/index.adoc]: I don't think we should document the intent to unify the Cluster resource. That initiative is aimed at helping cloud and we don't want to encourage self hosted users to start using the Cluster CRD. 

...We don't document anything about the Cluster or Console CRD for end users aside from this migration, right?

Comment (JakeSCahill) [reply modules/upgrade/pages/deprecated/index.adoc]: Thanks, Chris. Updated.

**Review by chrisseto — APPROVED** [review]
LGTM!

File: modules/upgrade/pages/deprecated/index.adoc
Comment (Feediver1) [reply modules/upgrade/pages/deprecated/index.adoc]: Thanks for this update. Docs does not ever announce future plans/updates, so good catch and glad to see it was removed.

File: modules/upgrade/pages/deprecated/index.adoc
Comment (Feediver1) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
|The deprecated Cluster and Console custom resources are no longer migrated to the new Flux-based version.
```

**Review by Feediver1 — APPROVED** [review]
minor suggestion

**Comment by JakeSCahill** [comment]
Waiting for a new operator release before merging. This change affects the new version of the operator.

**Pull request merged**
